### PR TITLE
[SPARK-47795][K8S][DOCS] Supplement the doc of job schedule for K8S

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR propose to supplement the doc of job schedule for K8S.

### Why are the changes needed?
Spark document missing the description of job schedule for K8S.

### Does this PR introduce _any_ user-facing change?
'No'.
Just update document.

### How was this patch tested?
Manual tests.

### Was this patch authored or co-authored using generative AI tooling?
'No'.

### CODE REVIEW

File: docs/job-scheduling.md
Comment (dongjoon-hyun) [inline docs/job-scheduling.md:56]: This line change is irrelevant to K8s.

File: docs/job-scheduling.md
Comment (dongjoon-hyun) [inline docs/job-scheduling.md:99]: This looks wrong to me because there is actually nothing to **follow** in the future work section. May I ask what do you want to say here?

**Review by dongjoon-hyun — COMMENTED** [review]
Is this a required change, @beliefer?

File: docs/job-scheduling.md
Comment (beliefer) [reply docs/job-scheduling.md:56]: Yes.
Because the link is not accurate even if it was not related to K8S.
Do you want me revert it?

**Comment by beliefer** [comment]
> Is this a required change, @beliefer?

I think we should add these document so as follows the other cluster mangers.

**Comment by beliefer** [comment]
cc @yaooqinn @LuciferYang

**Comment by yaooqinn** [comment]
I didn't even notice this page or section. The dropdown from the Navi Bar is enough for me.

Do we support scheduling jobs across applications? It's odd to me.

**Comment by yaooqinn** [comment]
Nit: Use K8s instead of K8S, the former is the official abbreviation

**Comment by beliefer** [comment]
> Do we support scheduling jobs across applications? It's odd to me.

This section is about scheduling across applications.
`Scheduling Within an Application` section is related to jobs.

File: docs/job-scheduling.md
Comment (beliefer) [reply docs/job-scheduling.md]: reverted.

File: docs/job-scheduling.md
Comment (dongjoon-hyun) [inline docs/job-scheduling.md:99]: Given the content, `### Caveats` section would be better than here.

File: docs/job-scheduling.md
Comment (yaooqinn) [inline docs/job-scheduling.md:59]:
```suggestion
  spark.kubernetes.executor.request.cores than spark.executor.cores. For more information, see the
```

File: docs/job-scheduling.md
Comment (yaooqinn) [inline docs/job-scheduling.md:58]:
```suggestion
  Spark on K8s offers higher priority versions of spark.kubernetes.executor.limit.cores and
```

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @beliefer .

**Comment by dongjoon-hyun** [comment]
Let's wait for @yaooqinn 's final sign-off, @beliefer .

File: docs/job-scheduling.md
Comment (bjornjorgensen) [inline docs/job-scheduling.md:103]: In [K8s mode](running-on-kubernetes.html), we cannot use this feature by setting `spark.shuffle.service.enabled` to `true` because Spark on K8s does not yet support the external shuffle service.

File: docs/job-scheduling.md
Comment (bjornjorgensen) [inline docs/job-scheduling.md:60]: spark.kubernetes.executor.limit.cores and
  spark.kubernetes.executor.request.cores than spark.executor.cores -> `spark.kubernetes.executor.limit.cores` and
  `spark.kubernetes.executor.request.cores` than `spark.executor.cores`

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
Merged to master

Thank you @beliefer @dongjoon-hyun @bjornjorgensen

**Comment by beliefer** [comment]
@yaooqinn @dongjoon-hyun @bjornjorgensen Thank you!

**Pull request closed without merging**
